virttest/qemu_devices/qcontainer.py

```diff
@@ -1718,6 +1719,146 @@ def images_define_by_params(self, name, image_params, media=None,
+                controller_suffix = 'device'
+            elif machine.startswith("s390"):
+                controller_suffix = 'ccw'
+            else:
+                controller_suffix = 'pci'
+            bus_type = 'virtio-serial-%s' % controller_suffix
+
+        prefix = params.get('virtio_port_name_prefix')
+        name = prefix + serial_id if prefix else serial_id
+        chardev_id = 'chardev_%s' % serial_id
+        params['chardev_name'] = params.get('chardev_name', serial_id)
+        chardev_device = self.chardev_define_by_params(chardev_id, params, file_name)
+        serial_devices = self.serials_define_by_variables(serial_id,
+                                                          serial_type,
+                                                          chardev_id,
+                                                          bus_type=bus_type,
+                                                          serial_name=name,
+                                                          bus=params.get(
+                                                              "serial_bus"),
+                                                          nr=params.get(
+                                                              "serial_nr"),
+                                                          reg=params.get(
+                                                              "serial_reg"))
+
+        return [chardev_device] + serial_devices
+
+    def chardev_define_by_params(self, chardev_id, params, file_name=None):
+        """
+        Wrapper for creating -chardev device from params.
+        :param chardev_id: chardev id
+        :param params: chardev params
+        :param file_name: file name of chardev (optional)
+        :return: CharDevice object
+        """
+        backend = params.get('chardev_backend', 'unix_socket')
+        # for tcp_socket and unix_socket, both form to 'socket'
+        _backend = 'socket' if 'socket' in backend else backend
+        # Generate -chardev device
+        chardev_param = Params({'backend': _backend, 'id': chardev_id})
+        if backend in ['unix_socket', 'file', 'pipe', 'serial',
+                       'tty', 'parallel', 'parport']:
+            chardev_param.update({'path': file_name})
+        elif backend in ['udp', 'tcp_socket']:
+            chardev_param.update(
+                {'host': params['chardev_host'],
+                 'port': params['chardev_port'],
+                 'ipv4': params.get('chardev_ipv4'),
+                 'ipv6': params.get('chardev_ipv6')})
+        if 'socket' in backend:  # tcp_socket & unix_socket
+            chardev_param.update(
+                {'server': params.get('chardev_server', 'yes'),
+                 'nowait': params.get('chardev_nowait', 'yes')})
+        elif backend in ['spicevmc', 'spiceport']:
+            chardev_param.update(
+                {'debug': params.get('chardev_debug'),
+                 'name': params.get('chardev_name')})
```

Comment:
same with before, test failed.

qemu command line:
```
    -chardev spicevmc,id=chardev_vs3,name=vs3 \
    -device virtserialport,id=vs3,name=com.redhat.spice.vs3,chardev=chardev_vs3,bus=virtio_serial_pci0.0,nr=3 \
```

failed reason:
```
qemu-kvm: -chardev spicevmc,id=chardev_vs3,name=vs3: unsupported type name: vs3
allowed spice char type names: vdagent, smartcard, usbredir'
```
I am not familiar with this,  `chardev_name` is must for `spicevmc` ?

Reply:
@yanan-fu Please refer here: https://github.com/autotest/tp-qemu/pull/1434/files#diff-8d0a580773b347958522648c9657235eL10 should be `vdagent`

Reply:
@vivianQizhu  Got the point.  The value of `name` option for `chardev` can only be `vdagent, smartcard, usbredir`.

In this way,  i think `params['chardev_name'] = params.get('chardev_name', serial_id)` is unreasonable.
If user want to test backend `spicevmc`,  `chardev_name` should be necessary, user should give correct value,  othervise, case failed as `qemu-kvm: -chardev spicevmc,id=chardev_vs3: chardev: spice channel: no name given'`

Reply:
@yanan-fu I think we better adopt the original usage(no `chardev_name`): https://github.com/avocado-framework/avocado-vt/pull/1639/files#diff-3f2317089f10636f453dca6c4f880f6cL1701, so by default user only need to define `serial_id` as `vdagent, smartcard, usbredir`, like previously.

Reply:
@yanan-fu Removed `params['chardev_name'] = params.get('chardev_name', serial_id)`